src/coreclr/vm/jitinterface.cpp

```diff
@@ -715,6 +715,71 @@ int CEEInfo::getStringLiteral (
     return result;
 }
 
+size_t CEEInfo::printObjectDescription (
+        void*  handle,
+        char*  buffer,
+        size_t bufferSize,
+        size_t* pRequiredBufferSize)
+{
+    CONTRACTL{
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    size_t bytesWritten = 0;
+
+    // NOTE: this function is used for pinned/frozen handles
+    // it doesn't need to null-terminate the string
+
+    _ASSERT(handle != nullptr && buffer != nullptr && bufferSize > 0);
+
+    JIT_TO_EE_TRANSITION();
+
+    Object* obj = (Object*)handle;
+
+    GCX_COOP();
+
+    StackSString stackStr;
+
+    // Currently only supported for String and RuntimeType
+    if (obj->GetMethodTable()->IsString())
+    {
+        ((StringObject*)obj)->GetSString(stackStr);
+    }
+    else if (obj->GetMethodTable() == g_pRuntimeTypeClass)
+    {
+        ((ReflectClassBaseObject*)obj)->GetType().GetName(stackStr);
+    }
+    else
+    {
+        _ASSERTE(!"Unexpected object type");
+    }
+
+    const UTF8* utf8data = stackStr.GetUTF8();
+    bytesWritten = min(bufferSize, stackStr.GetCount());
+    memcpy((BYTE*)buffer, (BYTE*)utf8data, bytesWritten);
+
+    if (bytesWritten == bufferSize)
+    {
+        // Trim data and null-terminate it
+        buffer[bytesWritten - 1] = 0;
+    }
+    else
+    {
+        buffer[bytesWritten] = 0;
+    }
+
+    if (pRequiredBufferSize != nullptr)
+    {
+        *pRequiredBufferSize = stackStr.GetCount();
```

Comment:
```suggestion
        *pRequiredBufferSize = stackStr.GetCount() + 1;
```
So that the output fits into the buffer.

Reply:
Ah, right, I had "real length" here in mind